src/github.com/matrix-org/dendrite/roomserver/input/events.go

```diff
@@ -79,8 +76,8 @@ func processRoomEvent(db RoomEventDatabase, ow OutputRoomEventWriter, input api.
 		if input.HasState {
 			// We've been told what the state at the event is so we don't need to calculate it.
 			// Check that those state events are in the database and store the state.
-			entries, err := db.StateEntriesForEventIDs(input.StateEventIDs)
-			if err != nil {
+			var entries []types.StateEntry
+			if entries, err = db.StateEntriesForEventIDs(input.StateEventIDs); err != nil {
```

Comment:
Hmm, I dunno I think it's fairly safe here TBH. You know the scope of entries from the `var entries` line and the variable shadowing checks prevent us from using `:=` if there is a `var entries` line.